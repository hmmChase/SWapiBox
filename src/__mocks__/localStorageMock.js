export default class LocalStorage {
  constructor() {
    this.store = {};
  }

  setItem(key, string) {
    this.store[key] = string;
  }

  getItem(key) {
    return this.store[key] || null;
  }

  removeItem(key) {
    delete this.store[key];
  }

  clear() {
    this.store = {};
  }
}

// global.localStorage = new LocalStorage();
