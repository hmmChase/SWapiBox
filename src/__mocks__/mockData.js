export const randomFilmCrawl = {
  crawl: 'Film Crawl',
  title: 'Movie Title',
  date: '1982-09-18'
};

export const cardObj1 = {
  name: 'T-16 skyhopper',
  model: 'T-16 skyhopper',
  class: 'repulsorcraft',
  passengers: '1'
};

export const cardObj2 = {
  name: 'Snowspeeder',
  model: 't-47 airspeeder',
  class: 'airspeeder',
  passengers: '0'
};

export const favorites = [cardObj1, cardObj2];

export const categoryData = [
  {
    name: 'Sand Crawler',
    model: 'Digger Crawler',
    class: 'wheeled',
    passengers: '30'
  },
  {
    name: 'T-16 skyhopper',
    model: 'T-16 skyhopper',
    class: 'repulsorcraft',
    passengers: '1'
  },
  {
    name: 'X-34 landspeeder',
    model: 'X-34 landspeeder',
    class: 'repulsorcraft',
    passengers: '1'
  },
  {
    name: 'TIE/LN starfighter',
    model: 'Twin Ion Engine/Ln Starfighter',
    class: 'starfighter',
    passengers: '0'
  },
  {
    name: 'Snowspeeder',
    model: 't-47 airspeeder',
    class: 'airspeeder',
    passengers: '0'
  },
  {
    name: 'TIE bomber',
    model: 'TIE/sa bomber',
    class: 'space/planetary bomber',
    passengers: '0'
  },
  {
    name: 'AT-AT',
    model: 'All Terrain Armored Transport',
    class: 'assault walker',
    passengers: '40'
  },
  {
    name: 'AT-ST',
    model: 'All Terrain Scout Transport',
    class: 'walker',
    passengers: '0'
  },
  {
    name: 'Storm IV Twin-Pod cloud car',
    model: 'Storm IV Twin-Pod',
    class: 'repulsorcraft',
    passengers: '0'
  },
  {
    name: 'Sail barge',
    model: 'Modified Luxury Sail Barge',
    class: 'sail barge',
    passengers: '500'
  }
];

export const fetchedFilmData = {
  opening_crawl: 'Film Crawl',
  title: 'Movie Title',
  release_date: '1982-09-18'
};

export const cleanFilmData = {
  crawl: fetchedFilmData.opening_crawl,
  title: fetchedFilmData.title,
  date: fetchedFilmData.release_date
};

export const fetchedPeopleData = {
  results: [
    {
      birth_year: '112BBY',
      created: '2014-12-10T15:10:51.357000Z',
      edited: '2014-12-20T21:17:50.309000Z',
      eye_color: 'yellow',
      films: (6)[
        ('https://swapi.co/api/films/2/',
        'https://swapi.co/api/films/5/',
        'https://swapi.co/api/films/4/',
        'https://swapi.co/api/films/6/',
        'https://swapi.co/api/films/3/',
        'https://swapi.co/api/films/1/')
      ],
      gender: 'n/a',
      hair_color: 'n/a',
      height: '167',
      homeworld: 'https://swapi.co/api/planets/1/',
      mass: '75',
      name: 'C-3PO',
      skin_color: 'gold',
      species: ['https://swapi.co/api/species/2/'],
      starships: [],
      url: 'https://swapi.co/api/people/2/',
      vehicles: []
    }
  ]
};

export const fetchedHomeWorldData = {
  climate: 'temperate',
  created: '2014-12-10T16:16:26.566000Z',
  diameter: '0',
  edited: '2014-12-20T20:58:18.452000Z',
  films: [],
  gravity: '1 standard',
  name: 'Tatooine',
  orbital_period: 'unknown',
  population: '200000',
  residents: ['https://swapi.co/api/people/10/'],
  rotation_period: 'unknown',
  surface_water: 'unknown',
  terrain: 'grass',
  url: 'https://swapi.co/api/planets/20/'
};

export const fetchedSpeciesData = {
  average_height: '180',
  average_lifespan: '120',
  classification: 'mammal',
  created: '2014-12-10T13:52:11.567000Z',
  designation: 'sentient',
  edited: '2015-04-17T06:59:55.850671Z',
  eye_colors: 'brown, blue, green, hazel, grey, amber',
  films: (7)[
    ('https://swapi.co/api/films/2/',
    'https://swapi.co/api/films/7/',
    'https://swapi.co/api/films/5/',
    'https://swapi.co/api/films/4/',
    'https://swapi.co/api/films/6/',
    'https://swapi.co/api/films/3/',
    'https://swapi.co/api/films/1/')
  ],
  hair_colors: 'blonde, brown, black, red',
  homeworld: 'https://swapi.co/api/planets/9/',
  language: 'Galactic Basic',
  name: 'Human',
  people: (35)[
    ('https://swapi.co/api/people/1/',
    'https://swapi.co/api/people/4/',
    'https://swapi.co/api/people/5/',
    'https://swapi.co/api/people/6/',
    'https://swapi.co/api/people/7/',
    'https://swapi.co/api/people/9/',
    'https://swapi.co/api/people/10/',
    'https://swapi.co/api/people/11/',
    'https://swapi.co/api/people/12/',
    'https://swapi.co/api/people/14/',
    'https://swapi.co/api/people/18/',
    'https://swapi.co/api/people/19/',
    'https://swapi.co/api/people/21/',
    'https://swapi.co/api/people/22/',
    'https://swapi.co/api/people/25/',
    'https://swapi.co/api/people/26/',
    'https://swapi.co/api/people/28/',
    'https://swapi.co/api/people/29/',
    'https://swapi.co/api/people/32/',
    'https://swapi.co/api/people/34/',
    'https://swapi.co/api/people/43/',
    'https://swapi.co/api/people/51/',
    'https://swapi.co/api/people/60/',
    'https://swapi.co/api/people/61/',
    'https://swapi.co/api/people/62/',
    'https://swapi.co/api/people/66/',
    'https://swapi.co/api/people/67/',
    'https://swapi.co/api/people/68/',
    'https://swapi.co/api/people/69/',
    'https://swapi.co/api/people/74/',
    'https://swapi.co/api/people/81/',
    'https://swapi.co/api/people/84/',
    'https://swapi.co/api/people/85/',
    'https://swapi.co/api/people/86/',
    'https://swapi.co/api/people/35/')
  ],
  skin_colors: 'caucasian, black, asian, hispanic',
  url: 'https://swapi.co/api/species/1/'
};

export const fetchedResidentsData = {
  name: 'Leia Organa',
  height: '150',
  mass: '49',
  hair_color: 'brown',
  skin_color: 'light',
  eye_color: 'brown',
  birth_year: '19BBY',
  gender: 'female',
  homeworld: 'https://swapi.co/api/planets/2/',
  films: [
    'https://swapi.co/api/films/2/',
    'https://swapi.co/api/films/6/',
    'https://swapi.co/api/films/3/',
    'https://swapi.co/api/films/1/',
    'https://swapi.co/api/films/7/'
  ],
  species: ['https://swapi.co/api/species/1/'],
  vehicles: ['https://swapi.co/api/vehicles/30/'],
  starships: [],
  created: '2014-12-10T15:20:09.791000Z',
  edited: '2014-12-20T21:17:50.315000Z',
  url: 'https://swapi.co/api/people/5/'
};

export const fetchedPlanetsData = {
  count: 61,
  next: 'https://swapi.co/api/planets/?page=2',
  previous: null,
  results: [
    {
      name: 'Alderaan',
      rotation_period: '24',
      orbital_period: '364',
      diameter: '12500',
      climate: 'temperate',
      gravity: '1 standard',
      terrain: 'grasslands, mountains',
      surface_water: '40',
      population: '2000000000',
      residents: ['https://swapi.co/api/people/5/'],
      films: ['https://swapi.co/api/films/6/', 'https://swapi.co/api/films/1/'],
      created: '2014-12-10T11:35:48.479000Z',
      edited: '2014-12-20T20:58:18.420000Z',
      url: 'https://swapi.co/api/planets/2/'
    }
  ]
};

export const cleanResidentsData = 'Leia Organa';

export const cleanPlanetsData = [
  {
    climate: 'temperate',
    name: 'Alderaan',
    population: '2000000000',
    residents: '',
    terrain: 'grasslands, mountains'
  }
];

export const fetchedVehicleData = {
  count: 39,
  next: 'https://swapi.co/api/vehicles/?page=2',
  previous: null,
  results: [
    {
      name: 'Sand Crawler',
      model: 'Digger Crawler',
      manufacturer: 'Corellia Mining Corporation',
      cost_in_credits: '150000',
      length: '36.8',
      max_atmosphering_speed: '30',
      crew: '46',
      passengers: '30',
      cargo_capacity: '50000',
      consumables: '2 months',
      vehicle_class: 'wheeled',
      pilots: [],
      films: ['https://swapi.co/api/films/5/', 'https://swapi.co/api/films/1/'],
      created: '2014-12-10T15:36:25.724000Z',
      edited: '2014-12-22T18:21:15.523587Z',
      url: 'https://swapi.co/api/vehicles/4/'
    }
  ]
};

export const cleanVehicleData = [
  {
    name: 'Sand Crawler',
    model: 'Digger Crawler',
    class: 'wheeled',
    passengers: '30'
  }
];

export const cleanHomeWorldData = {
  homeWorld: 'Tatooine',
  population: '200000'
};

export const cleanSpeciesData = {
  species: 'Human'
};

export const cleanPeopleData = [
  {
    homeWorld: undefined,
    name: 'C-3PO',
    population: undefined,
    species: undefined
  }
];
