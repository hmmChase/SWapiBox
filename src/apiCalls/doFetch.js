export const doFetch = async url => {
  try {
    const response = await fetch(url);
    if (!response.ok) {
      throw new Error(`${response.status}`);
    }
    return await response.json();
  } catch (error) {
    throw new Error(`Network request failed. (error: ${error.message})`);
  }
};
