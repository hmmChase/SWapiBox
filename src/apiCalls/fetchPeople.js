import { doFetch } from './doFetch';

export const fetchPeople = async () => {
  const root = 'http://swapi.com/api';
  const peopleData = await doFetch(`${root}/people`);
  return peopleData;
};
