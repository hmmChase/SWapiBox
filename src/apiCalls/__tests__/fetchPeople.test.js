import { fetchPeople, fetchHomeWorld, fetchSpecies } from '../fetchPeople';
import { doFetch } from '../doFetch';
jest.mock('../doFetch.js');
import * as mockData from '../../__mocks__/mockData';

describe('peopleData', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  describe('fetchPeople', () => {
    it('matches snapshot', async () => {
      await expect(fetchPeople).toMatchSnapshot();
    });

    it('calls doFetch', async () => {
      doFetch.mockImplementation(() =>
        Promise.resolve(mockData.fetchedPeopleData)
      );
      await fetchPeople();

      await expect(doFetch).toHaveBeenCalled();
    });

    it('returns people data if fetches are successful', async () => {
      doFetch.mockImplementation(() =>
        Promise.resolve(mockData.fetchedPeopleData)
      );

      await expect(fetchPeople()).resolves.toEqual(mockData.cleanPeopleData);
    });

    it('throws error if fetch fails', async () => {
      doFetch.mockImplementation(() => Promise.reject(new Error('mock error')));
      const expected = new Error('mock error');

      await expect(fetchPeople()).rejects.toEqual(expected);
    });
  });

  describe('fetchHomeWorld', () => {
    it('calls doFetch with correct args', async () => {
      const mockURL = 'https://swapi.co/api/mock';

      doFetch.mockImplementation(() =>
        Promise.resolve(mockData.fetchedHomeWorldData)
      );
      await fetchHomeWorld(mockURL);

      expect(doFetch).toHaveBeenCalledWith(mockURL);
    });

    it('returns HomeWorld data if fetch successful', async () => {
      doFetch.mockImplementation(() =>
        Promise.resolve(mockData.fetchedHomeWorldData)
      );

      await expect(fetchHomeWorld()).resolves.toEqual(
        mockData.cleanHomeWorldData
      );
    });

    it('throws error if fetch fails', async () => {
      doFetch.mockImplementation(() => Promise.reject(new Error('mock error')));
      const expected = new Error('mock error');

      await expect(fetchHomeWorld()).rejects.toEqual(expected);
    });
  });

  describe('fetchSpecies', () => {
    it('calls doFetch', async () => {
      doFetch.mockImplementation(() =>
        Promise.resolve(mockData.fetchedSpeciesData)
      );
      const mockURL = 'https://swapi.co/api/mock';
      await fetchSpecies(mockURL);

      expect(doFetch).toHaveBeenCalledWith(mockURL);
    });

    it('returns Species data', async () => {
      await doFetch.mockImplementation(() => mockData.fetchedSpeciesData);

      await expect(await fetchSpecies()).toEqual(mockData.cleanSpeciesData);
    });

    it('throws error if fetch fails', async () => {
      doFetch.mockImplementation(() => Promise.reject(new Error('mock error')));
      const expected = new Error('mock error');

      await expect(fetchSpecies()).rejects.toEqual(expected);
    });
  });
});
