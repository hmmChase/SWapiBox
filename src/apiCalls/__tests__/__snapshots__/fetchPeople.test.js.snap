// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`peopleData fetchPeople matches snapshot 1`] = `[Function]`;
